src/routes/player/JoinRoom.tsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { socket } from "../../socket";
+
+function JoinRoom() {
+    // Extract room_id from the URL
+    const { room_id: roomId } = useParams();
+
+    useEffect(() => {
+        // Check if room_id is available before emitting the message
+        if (roomId) {
+            // Emit a "joinRoom" event to the server with the roomId
+            socket.emit("joinRoom", roomId);
```

Comment:
[NIT] we should move all socket events to a shared WS_Event like we have on the backend